pkg/interceptors/github/github.go

```diff
@@ -0,0 +1,61 @@
+package github
+
+import (
+	"errors"
+	gh "github.com/google/go-github/github"
+	"net/http"
+
+	"github.com/tektoncd/triggers/pkg/interceptors"
+
+	triggersv1 "github.com/tektoncd/triggers/pkg/apis/triggers/v1alpha1"
+
+	"go.uber.org/zap"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type Interceptor struct {
+	KubeClientSet kubernetes.Interface
+	Logger        *zap.SugaredLogger
+	Github        *triggersv1.GithubInterceptor
+}
+
+func NewInterceptor(ei *triggersv1.EventInterceptor, k kubernetes.Interface, l *zap.SugaredLogger) interceptors.Interceptor {
+	return &Interceptor{
+		Logger:        l,
+		Github:        ei.Github,
+		KubeClientSet: k,
+	}
+}
+
+func getSecretToken(cs kubernetes.Interface, gh *triggersv1.GithubInterceptor) ([]byte, error) {
+	secret, err := cs.CoreV1().Secrets(gh.SecretRef.Namespace).Get(gh.SecretRef.SecretName, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	return secret.Data[gh.SecretRef.SecretKey], nil
+}
+
+func (w *Interceptor) ExecuteTrigger(payload []byte, request *http.Request, _ *triggersv1.EventListenerTrigger, _ string) ([]byte, error) {
+	// No secret set, just continue
+	if w.Github.SecretRef == nil {
```

Comment:
If the `GitHub` interceptor has been configured, shouldn't it require a `SecretRef`?

https://github.com/tektoncd/triggers/pull/221/files#diff-559661e0e83bb92fa19d37cc32078347R143

Reply:
Ah yeah, I think I was future proofing it a bit since there are other fields I want to add here.